backend/unparsed_value.go

```diff
@@ -69,21 +70,15 @@ func ParseVariableValues(vv map[string]UnparsedVariableValue, decls map[string]*
 		if !declared {
 			switch val.SourceType {
 			case terraform.ValueFromConfig, terraform.ValueFromAutoFile, terraform.ValueFromNamedFile:
-				// These source types have source ranges, so we can produce
-				// a nice error message with good context.
-				//
-				// This one is a warning for now because there is an existing
-				// pattern of providing a file containing the superset of
-				// variables across all configurations in an organization. This
-				// is deprecated in v0.12.0 because it's more important to give
-				// feedback to users who make typos. Those using this approach
-				// should migrate to using environment variables instead before
-				// this becomes an error in a future major release.
-				if seenUndeclaredInFile < 3 {
+				// We allow undeclared names for variable values from files and warn in case
+				// users have forgotten a variable {} declaration or have a typo in their var name.
+				// Some users will actively ignore this warning because they use a .tfvars file
+				// across multiple configurations.
+				if seenUndeclaredInFile < 2 {
 					diags = diags.Append(tfdiags.Sourceless(
 						tfdiags.Warning,
 						"Value for undeclared variable",
-						fmt.Sprintf("The root module does not declare a variable named %q but a value was found in file %q. To use this value, add a \"variable\" block to the configuration.\n\nUsing a variables file to set an undeclared variable is deprecated and will become an error in a future release. If you wish to provide certain \"global\" settings to all configurations in your organization, use TF_VAR_... environment variables to set these instead.", name, val.SourceRange.Filename),
+						fmt.Sprintf(strings.TrimSpace(undeclaredVariableWarning), name, val.SourceRange.Filename),
```

Comment:
Since this is a diagnostic rather than just a direct print to the terminal, the diagnostic renderer will automatically wrap it to fit the width of the terminal and so you shouldn't need to hard-wrap it like this.

(Further, I suspect that with this hard-wrapping it'll end up looking odd if you make your terminal narrower than 78 characters, because it'll include both the hard-wrapping and the automatically-calculated wrapping. :confounded: )

Reply:
Thanks for this info! It does automatically wrap. What I was bothered by is that when you shrink the size of the window once it's printed, the lines look weird because the text flows in such a way to break that side border like:

```
| Warning: This is a serious warning and we
are going to tell you
| about this serious warning which is very 
| serious
```

I'll change this back to the regular string business, but ☝️ is the separate issue I was thinking about

Reply:
Indeed, that's one unfortunate downside of using spatial formatting in our output, rather than just flowing text. I assume the same would happen with it hard-wrapped at 78 characters, but I guess it's less likely that someone would shrink their terminal smaller than that. :thinking:

Hopefully we can continue to improve on the diagnostic rendering in future releases, in a way that will improve all of the diagnostics systematically. One thing I've been thinking about and which your observation here might also suggest is that it might be good to still have an upper limit on line length while wrapping, in recognition of the fact that lines over a certain visual length are hard to scan, whereas today we're just accepting whatever width the terminal is set at even if it's obnoxiously wide.

Reply:
Yes! I think the latter point is something I can very much get behind -- and what I would think would be the motivation behind hard wrapping. This is one of those things that's inconsistent across Terraform, between help text, diags, UI output... some sort of design heuristic around it in the future would be neat (a not-really-corner case might be streamed apply output, where those lines, because they are filled in with variables, can get quite long, and I think nice not to wrap? or to think about how we can make them more concise 🤔 )Anyway, all a separate topic :)